api4/cloud.go

```diff
@@ -60,7 +60,17 @@ func (api *API) InitCloud() {
 	api.BaseRoutes.Cloud.Handle("/delete-workspace", api.APISessionRequired(selfServeDeleteWorkspace)).Methods(http.MethodDelete)
 }
 
+func ensureCloudInterface(c *Context, where string) {
```

Comment:
We'll need to return some sort of value, likely a boolean, from `ensureCloudInterface` so that the *caller* can check whether it needs to stop execution. Otherwise,
* the caller calls `ensureCloudInterface`
* `ensureCloudInterface` places the error in context
* The caller resumes execution until it errors (or more likely, panics)

Reply:
+1 on this approach